src/Compilers/CSharp/Portable/CodeGen/EmitStackAllocInitializer.cs

```diff
@@ -30,11 +30,22 @@ private void EmitStackAllocInitializers(TypeSymbol type, BoundArrayInitializatio
             else
             {
                 ImmutableArray<byte> data = this.GetRawData(initExprs);
-                _builder.EmitStackAllocBlockInitializer(data, inits.Syntax, _diagnostics);
+                if (data.All(datum => datum == data[0]))
```

Comment:
How are we assured that data is non empty here?

Reply:
The codepath is unreachable for initializers for less than two elements.